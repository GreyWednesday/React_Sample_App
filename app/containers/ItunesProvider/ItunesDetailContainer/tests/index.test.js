import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl, renderProvider, timeout } from '@utils/testUtils';
import { ItunesDetailContainerTest as ItunesDetailContainer, mapStateToProps } from '../index';

describe('<ItunesDetailContainer />', () => {
  let submitSpy;
  let match;
  let trackDetails;

  beforeEach(() => {
    submitSpy = jest.fn();
    match = {
      params: {
        trackId: 1469577741
      }
    };

    trackDetails = {
      wrapperType: 'track',
      kind: 'song',
      artistId: 909253,
      collectionId: 1469577723
    };
  });

  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(
      <ItunesDetailContainer match={match} dispatchRequestSongDetail={() => {}} dispatchClearTrackDetails={() => {}} />
    );
    expect(baseElement).toMatchSnapshot();
  });

  it('should dispatch requestSongDetail action when it is rendered', async () => {
    const { baseElement } = renderWithIntl(
      <ItunesDetailContainer match={match} dispatchRequestSongDetail={submitSpy} dispatchClearTrackDetails={() => {}} />
    );
    expect(baseElement).toBeTruthy();
    await timeout(200);
    expect(submitSpy).toBeCalled();
  });

  it('should render if it receives track details', () => {
    const { getByTestId } = renderProvider(
      <ItunesDetailContainer
        match={match}
        trackDetails={trackDetails}
        dispatchRequestSongDetail={() => {}}
        dispatchClearTrackDetails={() => {}}
      />
    );
    expect(getByTestId('detail-panel')).toBeTruthy();
  });

  it('should not crash if it receives no track details', () => {
    const { getByTestId } = renderProvider(
      <ItunesDetailContainer
        match={match}
        dispatchRequestSongDetail={() => {}}
        dispatchClearTrackDetails={() => {}}
        isLoading={false}
      />
    );
    expect(getByTestId('detail')).toBeTruthy();

    expect(getByTestId('song_not_found')).toHaveTextContent('Song Not Found');
  });

  it('should receive the selectSongs() selector', () => {
    expect(mapStateToProps({}).songs).toBeTruthy();
  });

  it('should receive the trackDetails() selector', () => {
    expect(mapStateToProps({}).trackDetails).toBeNull();
  });
});
