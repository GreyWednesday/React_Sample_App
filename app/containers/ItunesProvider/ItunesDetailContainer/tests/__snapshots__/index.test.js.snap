// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItunesDetailContainer /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ItunesDetailContainer__Wrapper-sc-1pgd82j-0 cQiwvY"
      data-testid="detail"
    >
      <p>
        <p
          class="T__StyledText-gjlic1-0 kYcyRq"
          data-testid="t"
        >
          Song Not Found
        </p>
      </p>
    </div>
  </div>
</body>
`;
